util/abs/provision_pe_xl_nodes.rb

```diff
@@ -87,13 +87,36 @@ def create_pe_xl_bolt_files(hosts, output_dir)
 #   hosts = provision_hosts_for_roles(roles)
 #   create_nodes_yaml(hosts, output_dir)
 def create_nodes_yaml(hosts, output_dir)
-  File.open("#{output_dir}/nodes.yaml", "w+") do |f|
-    f.puts NODES_YAML_START
-    hosts.each do |host|
-      hostname = host[:hostname]
-      f.puts "      - #{hostname}"
-    end
+  yaml = YAML.safe_load NODES_YAML
+  nodes = []
+
+  hosts.each do |host|
+    nodes << host[:hostname]
   end
+
+  yaml["groups"][0]["nodes"] = nodes
+  File.write("#{output_dir}/nodes.yaml", YAML.dump(yaml))
+end
+
+# Checks the nodes.yaml file to ensure it has been written correctly
+#
+# @author Bill Claytor
+#
+# @param [String] output_dir The directory where the file should be written
```

Comment:
Since you aren't outputting in this method, the name doesn't make sense.  Should be like yaml_dir or something.  And any reason to lock the name of the file to be nodes.yaml?  Like why not take a file path as an input and just check it.  Then this method isn't arbitrarily hardcoding the file name.

Reply:
Updated